virtual_rainforest/core/grid.py

```diff
@@ -232,14 +261,39 @@ def __init__(
             raise ValueError(f"The grid_type {self.grid_type} is not defined.")
 
         # Run the grid creation
-        self.cell_dict = creator(
+        grid_structure = creator(
             cell_area=self.cell_area,
             cell_nx=self.cell_nx,
             cell_ny=self.cell_ny,
             xoff=self.xoff,
             yoff=self.yoff,
         )
 
+        self.cell_id = grid_structure.ids
+        self.polygons = grid_structure.polygons
```

Comment:
I'm not sure of the utility of defining a dedicated class for grid structure and then, immediately after using it, get rid of it by unpacking its values. You could equally do:

```python
self.cell_id, self.polygons = creator(...)
```
With the same effect and less complexity.

Reply:
I see the point about the transient class. I wanted a consistent return type that should be used by anything registered as a grid creator. I first went with a custom Type as a tuple of ids and polys, but switched to NamedTuple to use named attributes not indices.

We _could_ just use that GridStructure internally, but I was trying to avoid the extra attribute depth (`grid.id` not `grid.gridstructure.id`). The suggestion above is certainly more elegant.

Reply:
I'm not convinced - having a transient class is, in the end, increasing code complexity and overheads while types are totally transparent and innocuous at runtime. Specially being so shot lived! What's the downside of using a custom signature used by all grid creators?

```python
GRID_STRUCTURE_SIG = Tuple[Tuple[int, ...], Tuple[Polygon, ...]]
```

Moreover, you can modify the `register_grid` decorator such that the validation of the length of the tuples takes place there.

But, anyway, it's up to you. It is not a huge deal. In general, custom classes should only be used when there are obvious gains over using built-in (or third party) solutions out of the box. I don't see that here.

Reply:
That's pretty much exactly where I started 😄 and then then I got drawn into `object.id` access, which really isn't needed. The only useful thing that a custom `dataclass` _would_ uniquely allow is to populate a default `ids` sequence but that really doesn't seem worth the extra complexity! I will revert back to the custom sig.

Reply:
Is there any particular value in checking the tuples are of equal length in the `register_grid` decorator rather than in the `Grid` class? We'd have to wrap the function call to update the decorator and it's a one liner within `Grid`.

Reply:
No advantage, really

Reply:
Right. I've committed those changes to the wrong PR - they are now sitting on the end of #94.

Having said that, I pulled the most recent change on this branch (f57049fb881927e5ebd4753007d2c1ea4322925e) into #94 to get the web build working. I _think_ the net result is that we can merge #94 and then simply close this one. That's irritatingly messy, but probably easier than `revert`ing the last commit to #94 and moving that commit onto this PR?

Reply:
@dalonsoa - IIRC you had some `zsh` magic that looked like it was including colorised branch names in the shell prompt. What was doing that?

Reply:
This is what you're looking for, although I use `fish` instead of `zsh`:

- https://medium.com/@elviocavalcante/5-steps-to-improve-your-terminal-appearance-on-mac-osx-f58b20058c84
- https://ostechnix.com/oh-fish-make-shell-beautiful/